api/datacontext/config/featuregates/type.go

```diff
@@ -0,0 +1,69 @@
+package featuregates
+
+import (
+	cfgcpi "ocm.software/ocm/api/config/cpi"
+	"ocm.software/ocm/api/datacontext/attrs/featuregatesattr"
+	"ocm.software/ocm/api/utils/runtime"
+)
+
+const (
+	ConfigType   = "featuregates" + cfgcpi.OCM_CONFIG_TYPE_SUFFIX
+	ConfigTypeV1 = ConfigType + runtime.VersionSeparator + "v1"
+)
+
+func init() {
+	cfgcpi.RegisterConfigType(cfgcpi.NewConfigType[*Config](ConfigType, usage))
+	cfgcpi.RegisterConfigType(cfgcpi.NewConfigType[*Config](ConfigTypeV1, usage))
+}
+
+type FeatureGate = featuregatesattr.FeatureGate
+
+// Config describes a memory based repository interface.
+type Config struct {
```

Comment:
Whats the reason to have a config and an attribute set, shouldnt attributes alone be enough?